src/test/java/org/jenkinsci/plugins/workflow/WorkflowJobNonRestartingTest.java

```diff
@@ -236,4 +238,25 @@ public void addAction() throws Exception {
         p.addAction(a);
         assertNotNull(p.getAction(WhoAmI.class));
     }
+
+    @Test
+    public void killInfiniteLoop() throws Exception {
+        p.setDefinition(new CpsFlowDefinition(WorkflowJobNonRestartingTest.class.getName()+".going(); while (true) {}", true));
+
+        QueueTaskFuture<WorkflowRun> f = p.scheduleBuild2(0);
+        WorkflowRun b = f.getStartCondition().get(3, TimeUnit.SECONDS);
+
+        going.block(3000);    // get the buld going, which will loop infinitely
```

Comment:
There is a race condition here, that the test thread might continue after `going` is called but before the loop is entered. Whether that matters or not, I am not sure.